pkg/reconciler/tenancy/clusterworkspaceshard/clusterworkspaceshard_controller.go

```diff
@@ -83,16 +85,19 @@ func (c *Controller) enqueue(obj interface{}) {
 		runtime.HandleError(err)
 		return
 	}
-	klog.Infof("queueing workspace shard %q", key)
+	klog.Background().WithValues(logging.ReconcilerKey, controllerName, logging.KeyKey, key).Info("queueing workspace shard")
```

Comment:
Unclear if we ever really need logs here (that we're enqueuing) given that we should be safe to assume that we de-queue at some point.... but in any case there's no context provided to the controller's constructor, so we can't share the logger.

Reply:
We have had cases where we missed events from the informers. And we had cases with backoffs. Having these has had value.